copy/copy.go

```diff
@@ -574,14 +584,16 @@ func (ic *imageCopier) copyBlobFromStream(srcStream io.Reader, srcInfo types.Blo
 	isCompressed := decompressor != nil
 
 	// === Report progress using a pb.Reader.
-	bar := pb.New(int(srcInfo.Size)).SetUnits(pb.U_BYTES)
-	bar.Output = ic.reportWriter
-	bar.SetMaxWidth(80)
-	bar.ShowTimeLeft = false
-	bar.ShowPercent = false
-	bar.Start()
-	destStream = bar.NewProxyReader(destStream)
-	defer bar.Finish()
+	if ic.hasReportWriter {
```

Comment:
If making the progress bar conditional is that important for performance, this needs a comment explaining why it is conditional; otherwise in a few months someone will notice that with an `ioutil.Discard` writer implementation the conditional is redundant and remove it again.